modules/apps/document-library/document-library-service/src/main/java/com/liferay/document/library/internal/configuration/DLPreviewLimitConfigurationProviderImpl.java

```diff
@@ -0,0 +1,279 @@
+			_updateGroupPreviewLimit(scopePK, maxFileSize, maxNumberOfPages);
+		}
+		else if (scope == ExtendedObjectClassDefinition.Scope.SYSTEM) {
+			_updateSystemPreviewLimit(maxFileSize, maxNumberOfPages);
+		}
+		else {
+			throw new IllegalArgumentException("Unsupported scope: " + scope);
+		}
+	}
+
+	private <T> T _executeOne(
+		ExtendedObjectClassDefinition.Scope scope, long input,
+		Function<Long, T> companyFunction, Function<Long, T> groupFunction,
+		Supplier<T> systemFunction) {
+
+		if (scope == ExtendedObjectClassDefinition.Scope.COMPANY) {
+			return companyFunction.apply(input);
+		}
+		else if (scope == ExtendedObjectClassDefinition.Scope.GROUP) {
+			return groupFunction.apply(input);
+		}
+		else if (scope == ExtendedObjectClassDefinition.Scope.SYSTEM) {
+			return systemFunction.get();
+		}
+
+		throw new IllegalArgumentException("Unsupported scope: " + scope);
+	}
+
+	private long _getGroupMaxFileSizeLimit(long groupId) {
+		Group group = _groupLocalService.fetchGroup(groupId);
+
+		if (group == null) {
+			return getSystemMaxFileSize();
+		}
+
+		return getCompanyMaxFileSize(group.getCompanyId());
+	}
+
+	private int _getGroupMaxNumberOfPagesLimit(long groupId) {
+		Group group = _groupLocalService.fetchGroup(groupId);
+
+		if (group == null) {
+			return getSystemMaxNumberOfPages();
+		}
+
+		return getCompanyMaxNumberOfPages(group.getCompanyId());
+	}
+
+	private boolean _isLimitExceeded(
+		long limit, long value, long unlimitedValue) {
+
+		if ((limit != unlimitedValue) &&
+			((value == unlimitedValue) || (limit < value))) {
+
+			return true;
+		}
+
+		return false;
+	}
+
+	private void _updateCompanyPreviewLimit(
+			long companyId, long maxFileSize, int maxNumberOfPages)
+		throws Exception {
+
+		_validateMaxFileSize(maxFileSize, getSystemMaxFileSize());
+		_validateMaxNumberOfPages(
+			maxNumberOfPages, getSystemMaxNumberOfPages());
+
+		_dlPreviewLimitConfigurationHelper.updateScopedConfiguration(
+			maxFileSize, maxNumberOfPages,
+			ExtendedObjectClassDefinition.Scope.COMPANY, companyId);
+	}
+
+	private void _updateGroupPreviewLimit(
+			long groupId, long maxFileSize, int maxNumberOfPages)
+		throws Exception {
+
+		_validateMaxFileSize(maxFileSize, getSystemMaxFileSize());
+		_validateMaxNumberOfPages(
+			maxNumberOfPages, getSystemMaxNumberOfPages());
+
+		Group group = _groupLocalService.getGroup(groupId);
+
+		_validateMaxFileSize(
+			maxFileSize, getCompanyMaxFileSize(group.getCompanyId()));
+		_validateMaxNumberOfPages(
+			maxNumberOfPages, getCompanyMaxNumberOfPages(group.getCompanyId()));
+
+		_dlPreviewLimitConfigurationHelper.updateScopedConfiguration(
+			maxFileSize, maxNumberOfPages,
+			ExtendedObjectClassDefinition.Scope.GROUP, groupId);
+	}
+
+	private void _updateSystemPreviewLimit(
+			long maxFileSize, int maxNumberOfPages)
+		throws Exception {
+
+		_dlPreviewLimitConfigurationHelper.updateSystemConfiguration(
+			maxFileSize, maxNumberOfPages);
+	}
```

Comment:
Shouldn't this be named `_updateSystemPreviewConfiguration`? Better, inline it, as it is used only once (and it is a one liner).